### ISSUE TITLE
VF-IS Upgrade from 0.6.3 to 0.12.9b2 does not load the existing DB on Ubuntu 16.04

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri-installer-debian#84]
### Observed behavior
Kolibri was run without issues as 0.6.3 PEX on Ubuntu, DB had 10 users who all had some interactions with content, 2 classes each with an exam tried by at least one learner.

![0 6 3-VF](https://user-images.githubusercontent.com/1457929/68356191-e79b2e00-0111-11ea-9879-5a8a1afff077.gif)

Installing the 0.12.9b2 from a DEB file was uneventful, but once re-loaded in the browser, sign in page appears for a second and then gets replaced with the setup wizard, as if the existing 0.6.3 DB was not there. You can see in the screncast that a `db.sqlite3` file & Co appear, so my suspicion is that it somehow takes _precedence_ over the `instant_school.db` and `instant_schools_primary.sqlite3`.

Note also that no `backup` folder is created, which is what should happen when the version upgrade is happening.

![upgrade from 0 6 3 to 0 12 9 VF](https://user-images.githubusercontent.com/1457929/68356421-9b042280-0112-11ea-833e-fdfd7ccb872c.gif)

This is specific to Ubuntu, as on Windows 7 the DB from 0.6.4b1 is correctly recognized and all the users can see their previous progress once the upgrade to 0.12.9b2 has been performed.

### Expected behavior
0.6.3 interactions and progress should be available after the upgrade to 0.12.9.

### User-facing consequences
Loss of past progress on the platform.

### Errors and logs
…

### Steps to reproduce
Upgrade from 0.6.3 to 0.12.9

### Context
 * Kolibri version: 0.6.3 PEX upgraded to 0.12.9b2 DEB
 * Operating system: Ubuntu 16.04
 * Browser: all

**Comment by benjaoming** [comment]
@jredrejo this issue is a bit loaded with info, and I can't make out this part: Where is the intended settings module defined? It seems like `kolibri_instant_schools_plugin.instant_schools_settings` is the intended setting?

**Comment by jredrejo** [comment]
This is where the plugin defines it:
https://github.com/learningequality/kolibri-instant-schools-plugin/blob/02e183d8268a5811dcc2d923b073185408073e0a/kolibri_instant_schools_plugin/instant_schools_settings.py
but the `/etc/default/kolibri`file overrides it, so the database is changed and the upgrade is broken.
My question is: why is the Debian package setting this env variable? (and the others with a comment for Ubuntu 14.04 too, but this one is the important one in this case)

**Comment by benjaoming** [comment]
Variables can be defined in `/etc/kolibri/conf.d/<foo>.conf` - but on Ubuntu 14.04, there was an issue about making those environment variables readable in the subprocess. You can use `git blame`.

> This is where the plugin defines it:

Yup that looks like a custom settings module :) To use a custom settings module for the Debian system service, one can do:

```
sudo mkdir -p /etc/kolibri/conf.d
sudo echo 'DJANGO_SETTINGS_MODULE="kolibri_instant_schools_plugin.instant_schools_settings"' > /etc/kolibri/conf.d/vodafone.conf
```

What I still don't understand is: How did this custom settings module get passed to Kolibri before?

**Comment by jredrejo** [comment]
the thing is that if you dont set that variable, Kolibri uses the default. When the plugin is used, the default is provided by the plugin, unless you change it, as it is happening with the Debian package.
So my question is, what is the problem in removing that env variable from the `default`f ile that's causing problems ? 
As far as I see options.ini replaces it and , in case someone needs it, the Debian package is overwritting it in https://github.com/learningequality/kolibri-installer-debian/blob/master/debian/startup/kolibri.default#L27 , thus breaking any user previous setting.

**Comment by benjaoming** [comment]
> the thing is that if you dont set that variable, Kolibri uses the default. When the plugin is used, the default is provided by the plugin

Can you refer to this particular behavior in the plugin code so we're on the same page?

**Comment by rtibbles** [comment]
It's not the plugin code, it's Kolibri's behaviour for setting the default value for `DJANGO_SETTINGS_MODULE`: https://github.com/learningequality/kolibri/blob/3a29c516669139678fdacdb14c9a3a440b5b1994/kolibri/utils/env.py#L74

The custom build pipeline allows this to be set to a different value, in order to embed different default settings.

As such, the Debian behaviour here is, in the benign case, superfluous, and in this case causing a bother.

**Comment by benjaoming** [comment]
Before calling it "superfluous", maybe ask if it was there for a reason?

**Comment by jredrejo** [comment]
@benjaoming yes, please, can you explain me the rationale behind https://github.com/learningequality/kolibri-installer-debian/blob/master/debian/startup/kolibri.default#L27 
I can understand the `export`commands if you want to have variables available in `conf.d` but I can't understand why this one is set with that value that overrides any user or kolibri code setting.

**Comment by benjaoming** [comment]
My theory of why it was done is this:

1) Issuing `export DJANGO_SETTINGS_MODULE` in [L56](https://github.com/learningequality/kolibri-installer-debian/blob/master/debian/startup/kolibri.default#L56) causes an empty environment variable, which failed. Therefore a default was ensured in [L27](https://github.com/learningequality/kolibri-installer-debian/blob/master/debian/startup/kolibri.default#L27). It was necessary to write all these export statements in cases where people defined the environment variable differently in `conf.d` while running Ubuntu 14.04. All these export statements are, however, impossible to maintain properly, and the demand is low, since 14.04 is not very widely used.

2) It was properly assumed that the Debian package is allowed to create its own environment. To me, there is no principle stopping that the outermost layer defines an environment. It's not normal to configure the environment from within the application, that's why it's called an _environment_ :)

What I would propose is this:

1. Short term: We drop 14.04 support. I get rid of all the export statements and the definition of `DJANGO_SETTINGS_MODULE` in the system service.

2. Long term: We figure out how to stop defining settings environments from inside plugins (still not sure where it takes place?). This assumption will keep breaking and keep causing issues down the road.

**Comment by rtibbles** [comment]
We're not defining an environment from a plugin. We're defining it during build - we set it from Python code in the main Kolibri `set_env` function, because that is what is universal to all Kolibri instances.

Also, this is just about defining the default, the environment is still free to overwrite it, it just shouldn't also set a default that may get out of sync with the default set by Kolibri.

**Comment by benjaoming** [comment]
What build defines this? A code reference would be helpful.

**Comment by benjaoming** [comment]
> it just shouldn't also set a default that may get out of sync with the default set by Kolibri.

Agreed.

But I want to know how this plugin / special build defines the default environment to ensure that it doesn't break any other assumptions.

**Comment by rtibbles** [comment]
It gets set during the build by this https://github.com/learningequality/kolibri/blob/develop/build_tools/customize_build.py#L55

It gets picked up here at runtime https://github.com/learningequality/kolibri/blob/3a29c516669139678fdacdb14c9a3a440b5b1994/kolibri/utils/env.py#L74

**Comment by benjaoming** [comment]
Thanks @rtibbles - will try out a fix in 14.04 to see the effects and get back.

**Comment by jredrejo** [comment]
@benjaoming  would not it be easier if we just remove all those `exports` and simplify the package?

In case Ubuntu 14.04 needs a special setup for a special case I would prefer to have its own special debianization using the changelog to define the target release. Launchpad allows it and 14.04 won't break.

I don't see any benefit in adding more and  more code to support an old version that has one single use case. The simpler and less code the Debian package adds the better.
